packages/lib-user/src/components/shared/AllProjects/AllProjects.js

```diff
@@ -13,59 +19,109 @@ const StyledBox = styled(Box)`
+  const numPages = Math.ceil(numProjects / PAGE_SIZE)
+
+  // fetch project data from panoptes
+  const projectsQuery = {
+    cards: true,
+    id: projectIds?.join(','),
+    page_size: PAGE_SIZE
+  }
+
+  const {
+    renderedProjects,
+    error: projectsError,
+    isLoading: projectsLoading,
+    isValidating,
+    size,
+    setSize
+  } = useInfiniteScrollProjects(projectContributions, projectsQuery)
+
+  // load more projects into the UI
+  function handleLoadMore() {
+    if (size <= numPages) {
+      setSize(size + 1)
+    }
+  }
+
+  const loading = containerLoading || projectsLoading
+  const error = containerError || projectsError
+
+  const disableLoadMore =
+    loading || isValidating || error || numProjects === 0 || size === numPages
 
   return (
     <>
       {loading ? (
         <Box fill align='center' justify='center'>
           <Loader />
         </Box>
-      ) : projects?.length === 0 ? (
-        <Box fill align='center' pad='medium'>
-          <Text>{t('AllProjects.noProjects')}</Text>
-        </Box>
       ) : error ? (
         <Box fill align='center' pad='medium'>
           <Text>{t('AllProjects.error')}</Text>
         </Box>
+      ) : numProjects === 0 ? (
+        <Box fill align='center' pad='medium'>
+          <Text>{t('AllProjects.noProjects')}</Text>
+        </Box>
       ) : (
-        <StyledBox forwardedAs='ul' direction='row' wrap justify='center'>
-          {projects.map(project => {
-            return (
-              <li key={project?.id}>
-                <ProjectCard
-                  badge={project?.count}
-                  description={project?.description}
-                  displayName={project?.display_name}
-                  href={`https://www.zooniverse.org/projects/${project?.slug}`}
-                  imageSrc={project?.avatar_src}
-                  size='large'
-                />
-              </li>
-            )
-          })}
-        </StyledBox>
+        <>
+          <StyledBox forwardedAs='ul' direction='row' wrap justify='center'>
+            {renderedProjects?.map(project => {
+              return (
+                <li key={project?.id}>
+                  <ProjectCard
+                    badge={project?.count}
+                    description={project?.description}
+                    displayName={project?.display_name}
+                    href={`https://www.zooniverse.org/projects/${project?.slug}`}
+                    imageSrc={project?.avatar_src}
+                    size={grommetSize}
+                  />
+                </li>
+              )
+            })}
+          </StyledBox>
+          {loading || isValidating && size > 1 && (
+            <Box align='center' pad='medium'>
+              <Loader />
+            </Box>
+          )}
+          <Box align='center'>
+            <PrimaryButton
+              label={t('AllProjects.more')}
+              disabled={disableLoadMore}
+              onClick={handleLoadMore}
+            />
```

Comment:
Keyboard/screen reader accessibility might trip up here, in that focus will stay with this button after loading a fresh page of projects.

Reply:
That's a good point, and I plan to do some focus management around the Load More handler. For instance, for sighted users it'd be helpful to bring the first of the newly loaded projects into view.